rust/src/event.rs

```diff
@@ -239,6 +249,17 @@ mod test_event {
             },
             None => assert!(false),
         }
+
+        let web_vitals = get_web_vital();
+        let mut web_vitals_buffer: Vec<u8> = Vec::new();
+        web_vitals.encode(&mut web_vitals_buffer).unwrap();
+        match Event::decode(&EventModel::WebVitals, &*web_vitals_buffer) {
+            Some(event) => match event {
+                Event::WebVitals(web_vital_event) => assert_eq!(web_vital_event, web_vitals),
+                _ => assert!(false),
+            },
+            None => assert!(false),
```

Comment:
I think you could destructure `Event::WebVitals` in the first match:
```suggestion
        match Event::decode(&EventModel::WebVitals, &*web_vitals_buffer) {
            Some(Event::WebVitals(web_vital_event) => assert_eq!(web_vital_event, web_vitals),
            _ => assert!(false),
```